### PR TITLE
102.possibility to reload list of pupils

### PR DESCRIPTION
**Pull request by shagi9**

### CODE REVIEW

File: src/app/components/add-class-form/add-class-form.component.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export class AddClassFormComponent implements OnInit {
   constructor(
     private formBuilder: FormBuilder,
     private http: HttpClient,
-    private pupilService: PupilinfoService,
+    private pupilService: PupilsService,
```
Comment (DmMov) [inline src/app/components/add-class-form/add-class-form.component.ts]: The variable is still private and named pupilService

File: src/app/components/add-pupil-form/add-pupil-form.component.ts [hunk 2]
```diff
@@ -24,7 +24,8 @@ export class AddPupilFormComponent implements OnInit {
   constructor(
     private formBuilder: FormBuilder,
     private http: HttpClient,
-    private httpOptionsService: HttpOptionsService
+    private httpOptionsService: HttpOptionsService,
+    private pupilsService : PupilsService
```
Comment (DmMov) [inline src/app/components/add-pupil-form/add-pupil-form.component.ts:28]: Here it should be private, but you added an extra space.

File: src/app/pages/pupils/pupils.component.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { PupilinfoService } from '../../services/pupils/pupilinfo.service';
 
 export class PupilsComponent implements OnInit {
 
-  constructor(private pupilService : PupilinfoService) {}
+  constructor(public pupilsService : PupilsService) {}
```
Comment (DmMov) [inline src/app/pages/pupils/pupils.component.ts]: extra space

File: src/app/services/pupils/pupils.service.ts [hunk 4]
```diff
@@ -2,29 +2,41 @@ import { get } from 'js-cookie';
...
-export class PupilinfoService {
+export class PupilsService {
+  pupils: IPupil[] = [];
   private url: string = "https://localhost:5001/api/Pupils";
 
-  constructor(private http: HttpClient, private httpOptions: HttpOptionsService) { }
+  constructor(
+    private http: HttpClient, 
+    private httpOptionsService: HttpOptionsService
+    ) { }
```
Comment (DmMov) [inline src/app/services/pupils/pupils.service.ts]: exptra tab

File: src/app/services/pupils/pupils.service.ts [hunk 5]
```diff
@@ -2,29 +2,41 @@ import { get } from 'js-cookie';
...
-    return this.http.get<IPupil>(this.url + "/GetPupilById", this.httpOptions.options);
+    this.httpOptionsService.loadHeaders();
+    return this.http.get<IPupil>(this.url + "/GetPupilById", this.httpOptionsService.options);
   }
   getPupilsToSelect = (numb: number): Observable<IPupilToSelect[]> => {
-    this.httpOptions.loadHeaders();
-    return this.http.get<IPupilToSelect[]>(this.url + "/GetPupilsBySchoolShort" + "?numbOfClass=" + numb, this.httpOptions.options);
+    this.httpOptionsService.loadHeaders();
+    return this.http.get<IPupilToSelect[]>(this.url + "/GetPupilsBySchoolShort" + "?numbOfClass=" + numb, this.httpOptionsService.options);
   }
```
Comment (DmMov) [inline src/app/services/pupils/pupils.service.ts]: Add an empty line beetwen these functions

File: src/app/components/table-user/table-user.component.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import { Component, OnInit, Input } from '@angular/core';
+import { PupilsService } from 'src/app/services/pupils/pupils.service';
```
Comment (DmMov) [inline src/app/components/table-user/table-user.component.ts]: Why did you add it?
You don't use it.

Comment (shagi9) [reply src/app/components/table-user/table-user.component.ts]: I deleted this in one of the commits or not(

**Review by DmMov — DISMISSED** [review]

**Review by DmMov — APPROVED** [review]

**Review by Karonuk — APPROVED** [review]

**Review by Hrabarchuk — APPROVED** [review]
